public/app/features/profile/ChangePasswordForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { PureComponent, MouseEvent } from 'react';
+import config from 'app/core/config';
+import { Button, LinkButton, LoadingPlaceholder } from '@grafana/ui';
+import { ChangePasswordFields } from 'app/core/utils/UserProvider';
+import { PasswordInput } from 'app/core/components/PasswordInput/PasswordInput';
+
+export interface Props {
+  isLoading: boolean;
+  onChangePassword: (payload: ChangePasswordFields) => void;
+}
+
+export interface State {
+  oldPassword: string;
+  newPassword: string;
+  confirmNew: string;
+}
+
+export class ChangePasswordForm extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      oldPassword: '',
+      newPassword: '',
+      confirmNew: '',
+    };
+  }
+
+  onOldPasswordChange(oldPassword: string) {
+    this.setState({ oldPassword });
+  }
+
+  onNewPasswordChange(newPassword: string) {
+    this.setState({ newPassword });
+  }
+
+  onConfirmPasswordChange(confirmNew: string) {
+    this.setState({ confirmNew });
+  }
+
+  render() {
+    const { oldPassword, newPassword, confirmNew } = this.state;
+    const { onChangePassword, isLoading } = this.props;
+    const { ldapEnabled, authProxyEnabled } = config;
+
+    if (isLoading) {
```

Comment:
Isn't this gonna make the entire form blink? Feels like a bad UX. I woul d suggest enabling/didabling the button according to `isLoading` state.